File: CppPractice/Project1/Project1/Leetcode155.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class MinStack {
public:
    /** initialize your data structure here. */
    vector<int> stack;
    MinStack() {
        MinStack::stack = stack;
    }

    void push(int val) {
        MinStack::stack.push_back(val);
    }

    void pop() {
        MinStack::stack.pop_back();
    }

    int top() {
        return MinStack::stack[MinStack::stack.size()-1];

    }

    int getMin() {
        int min_val = INT_MAX;
        for ( int i = 0; i < MinStack::stack.size(); i++)
        {
            if (min_val>MinStack::stack[i])
            {
                min_val = MinStack::stack[i];
            }
        }
        return min_val;
    }
};

File: Pythonleetcode/Leetcode155.py
class MinStack(object):

    def __init__(self):
        self.mystack=[]


    def push(self, val):
        self.mystack.append(val)


    def pop(self):
        self.mystack.pop()


    def top(self):
        return self.mystack[-1]


    def getMin(self):
        return min(self.mystack)
